### DISCUSSION TITLE
getting sibling/parent objects

### DISCUSSION THREAD

**Discussion by dragoncoder047** [kaplayjs/kaplay#297] [Q&A] [answered]
So I'm writing a custom component that will allow tiles in a level (from `addLevel()`) to communicate with each other based on tags.

Getting the children objects of the level is easy as pie, it's just `level.children`. That assumes I have a reference to the level. I do, but it doesn't seem right to rely on that, since I have to store it in a global variable to make it accessible to the component (the level hasn't been returned by `addLevel()` at the time the component is created).

If I *only* have a reference to a particular tile object, how would I find all of the siblings (or the tile's parent object, if that makes it any easier)?

**Comment by amyspark-ng** [comment, marked as answer]
You can use the query function, which is a more powerful get() function, with lots of functions
https://kaplayjs.com/doc/QueryOpt/

query({
    hierarchy: "siblings"
})
